SSA/Projects/InstCombine/AliveAutoGenerated.lean

```diff
@@ -1,11 +1,58 @@
 import SSA.Projects.InstCombine.LLVM.EDSL
-import SSA.Projects.InstCombine.Tactic
+import SSA.Projects.InstCombine.AliveStatements
+import Mathlib.Tactic
 open MLIR AST
 
+open Std (BitVec)
+
+namespace AliveAutoGenerated
+
 abbrev ICom.Refinement (src tgt : Com (φ:=0) Γ t) (h : Goedel.toType t = Option α := by rfl) : Prop :=
-  ∀ Γv, Bitvec.Refinement (h ▸ src.denote Γv) (h ▸ tgt.denote Γv)
+  ∀ Γv, (h ▸ src.denote Γv) ⊑ (h ▸ tgt.denote Γv)
 
-infixr:90 " ⊑ "  => ICom.Refinement
+infixr:90 (priority:=high) " ⊑ "  => ICom.Refinement
+
+/--
+- We first simplify `ICom.refinement` to see the context `Γv`.
+- We `simp_peephole Γv` to simplify context accesses by variables.
+- We simplify the translation overhead.
+- Then we introduce variables, `cases` on the variables to eliminate the `none` cases.
+- We cannot leave it at this state, since then the variables will be inaccessible.
+- So, we revert the variables for the user to re-introduce them as they see fit.
+-/
+macro "simp_alive_peephole" : tactic =>
```

Comment:
Would it make sense to move this to `SSA.Projects.InstCombine.Tactic` and name it simp_alive. The move and rename of this tactic hides that fact that `try simp only` is mostly the same?

Reply:
I would prefer us to keep it here, and honestly, find a better name, since we will have *two* tactics moving forward:
- `simp_alive_peephole` to eliminate the rewriter junk to get a pure theorem, and
- `alive_auto` or such, to try to be a decision procedure that tries `ring` and other such tactics to simplify theorems. 

I am not a fan of either of the names, since I find them confusing. To help bikeshedding, I would name the something like:
- `alive_cleanup`: this "cleans up" the state to get a pure maths theorem
- `alive_decide` this is a "decision procedure" that tries to simplify the proof as much as possible.

Reply:
Is there a specific reason you prefer a the tactic definition to be places in a file full of auto-generated theorems, rather than in its own tactic-specific file? That seems surprising to me.

I am happy with renaming the tactics, but find a complex PR the wrong place to do so. Again, this seems to be a great self-contained PR.